inlang/source-code/website/src/pages/blog/@id/index.page.tsx

```diff
@@ -1,31 +1,112 @@
 import { Meta, Title } from "@solidjs/meta"
 import { Layout } from "#src/pages/Layout.jsx"
-import { Markdown, parseMarkdown } from "#src/services/markdown/index.js"
-import { Show } from "solid-js"
-import type { ProcessedTableOfContents } from "./index.page.server.jsx"
+import { Show, onMount } from "solid-js"
+import type { GeneratedTableOfContents as ProcessedTableOfContents } from "./index.page.server.jsx"
+import { defaultLanguage } from "#src/renderer/_default.page.route.js"
+import { useI18n } from "@solid-primitives/i18n"
+import { currentPageContext } from "#src/renderer/state.js"
+import "@inlang/markdown/css"
+import "@inlang/markdown/custom-elements"
 
 /**
  * The page props are undefined if an error occurred during parsing of the markdown.
  */
 export type PageProps = {
 	processedTableOfContents: ProcessedTableOfContents
-	markdown: Awaited<ReturnType<typeof parseMarkdown>>
+	meta: ProcessedTableOfContents[string]
+	markdown: Awaited<ReturnType<any>>
 }
 
 export function Page(props: PageProps) {
+	const [, { locale }] = useI18n()
+
+	const getLocale = () => {
+		const language = locale() ?? defaultLanguage
+		return language !== defaultLanguage ? "/" + language : ""
+	}
+
+	const replaceChars = (str: string) => {
+		return str
+			.replaceAll(" ", "-")
+			.replaceAll("/", "")
+			.replace("#", "")
+			.replaceAll("(", "")
+			.replaceAll(")", "")
+			.replaceAll("?", "")
+			.replaceAll(".", "")
+			.replaceAll("@", "")
+			.replaceAll(/([\uE000-\uF8FF]|\uD83C[\uDF00-\uDFFF]|\uD83D[\uDC00-\uDDFF])/g, "")
+			.replaceAll("✂", "")
+	}
+
+	const scrollToAnchor = (anchor: string, behavior?: ScrollBehavior) => {
+		const element = document.getElementById(anchor)
+		if (element && window) {
+			window.scrollTo({
+				top: element.offsetTop - 96,
+				behavior: behavior ?? "instant",
+			})
+		}
+		window.history.pushState({}, "", `${currentPageContext.urlParsed.pathname}#${anchor}`)
+	}
+
+	onMount(async () => {
+		for (const heading of props.markdown
+			.match(/<h[1-3].*?>(.*?)<\/h[1-3]>/g)
+			.map((heading: string) => {
+				// We have to use DOMParser to parse the heading string to a HTML element
+				const parser = new DOMParser()
```

Comment:
What is all this code for? Why the replacements?

Reply:
This is the code of the tableOfContents and the code of scrolling to the specific anchors (at it is not working otherwise, -> seen yesterday). Also replace function necessary to sanitize the headings from emojis which others have been used in the readmes.